modules/kernel/mkosi.build.chroot

```diff
@@ -103,11 +103,11 @@ KERNEL_RELEASE=$(make $EXTRA -s kernelrelease)
 ln -sf "$BUILDDIR" /tmp/lib
 mkdir -p "$BUILDDIR/modules/$KERNEL_RELEASE"
 find "$BUILDDIR/modules" -mindepth 1 -maxdepth 1 ! -name "$KERNEL_RELEASE" -type d -exec rm -r {} +
-make $EXTRA INSTALL_MOD_PATH=/tmp modules_install
+make $EXTRA INSTALL_MOD_PATH="$DESTDIR" MODLIB="$DESTDIR/usr/lib/modules/$KERNEL_RELEASE" modules_install
 
 if ! ((INCREMENTAL)); then
     mkdir -p "$DESTDIR/usr/lib/modules/$KERNEL_RELEASE"
-    make $EXTRA INSTALL_MOD_PATH="$DESTDIR/usr" modules_install
+    make $EXTRA INSTALL_MOD_PATH="$DESTDIR" MODLIB="$DESTDIR/usr/lib/modules/$KERNEL_RELEASE" modules_install
```

Comment:
Ditto, if we're initializing `MODLIB` ourselves, there's no need to set `INSTALL_MOD_PATH`

Reply:
This is the SUSE-specific `depmod` issue I mention in the PR description. Just setting `MODLIB` or `INSTALL_MOD_PATH` doesn't work properly. [See this patch description for more details.](https://github.com/SUSE/kernel-source/blob/92abc10125a0cbecc93afb55f98e45e381e5fbbf/patches.rpmify/usrmerge-Adjust-module-path-in-the-kernel-sources.patch)

With `MODLIB="$DESTDIR/usr/lib/modules/$KERNEL_RELEASE"` you get this error:

```
  ...
  INSTALL /work/dest/usr/lib/modules/6.10.0-rc5/kernel/net/vmw_vsock/vsock.ko
  INSTALL /work/dest/usr/lib/modules/6.10.0-rc5/kernel/net/vmw_vsock/vsock_diag.ko
  INSTALL /work/dest/usr/lib/modules/6.10.0-rc5/kernel/net/vmw_vsock/vmw_vsock_virtio_transport.ko
  INSTALL /work/dest/usr/lib/modules/6.10.0-rc5/kernel/net/vmw_vsock/vmw_vsock_virtio_transport_common.ko
  INSTALL /work/dest/usr/lib/modules/6.10.0-rc5/kernel/net/vmw_vsock/vsock_loopback.ko
  DEPMOD  /work/dest/usr/lib/modules/6.10.0-rc5
depmod: ERROR: could not open directory /usr/lib/modules/6.10.0-rc5: No such file or directory
depmod: FATAL: could not search modules: No such file or directory
```

And with `INSTALL_MOD_PATH=$DESTDIR` you get these warnings:

```
depmod: WARNING: could not open modules.order at /work/dest/usr/lib/modules/6.10.0-rc5: No such file or directory
depmod: WARNING: could not open modules.builtin at /work/dest/usr/lib/modules/6.10.0-rc5: No such file or directory
depmod: WARNING: could not open modules.builtin.modinfo at /work/dest/usr/lib/modules/6.10.0-rc5: No such file or directory
```

And then later when `mkosi` does `modinfo` you get these errors:

```
‣  Running modinfo to fetch kernel module dependencies
modinfo: ERROR: Module vmw-vsock-virtio-transport not found.
modinfo: ERROR: Module cdrom not found.
modinfo: ERROR: Module nvmet-fc not found.
modinfo: ERROR: Module iptable-raw not found.
modinfo: ERROR: Module ip-gre not found.
modinfo: ERROR: Module xt-TCPMSS not found.
modinfo: ERROR: Module configs not found.
[the same for every module]
```

Setting both results in no warnings and the build works.

For reference, the original `INSTALL_MOD_PATH=$DESTDIR/usr` gives the following error because of the `/usr/lib` issue:

```
  ...
  INSTALL /work/dest/usr/lib/modules/6.10.0-rc5/kernel/net/vmw_vsock/vsock.ko
  INSTALL /work/dest/usr/lib/modules/6.10.0-rc5/kernel/net/vmw_vsock/vsock_diag.ko
  INSTALL /work/dest/usr/lib/modules/6.10.0-rc5/kernel/net/vmw_vsock/vmw_vsock_virtio_transport.ko
  INSTALL /work/dest/usr/lib/modules/6.10.0-rc5/kernel/net/vmw_vsock/vmw_vsock_virtio_transport_common.ko
  INSTALL /work/dest/usr/lib/modules/6.10.0-rc5/kernel/net/vmw_vsock/vsock_loopback.ko
  DEPMOD  /work/dest/usr/lib/modules/6.10.0-rc5
depmod: ERROR: could not open directory /work/dest/usr/usr/lib/modules/6.10.0-rc5: No such file or directory
depmod: FATAL: could not search modules: No such file or directory
```

<hr>

The SUSE patchset I linked suggests doing `MODLIB='$(INSTALL_MOD_PATH)/usr/lib/modules/$(KERNELRELEASE)'` to fix the problem. I can do that instead if you prefer:

```
make $EXTRA INSTALL_MOD_PATH="$DESTDIR" MODLIB='$(INSTALL_MOD_PATH)/usr/lib/modules/$(KERNELRELEASE)' modules_install
```

(The patch description says `INSTALL_MOD_PATH` but that's a typo -- you get a recursive reference that way and make refuses to build. Only `MODLIB='$(INSTALL_MOD_PATH)/usr/lib/modules/$(KERNELRELEASE)'` make sense.)

Reply:
Yeah let's do it like that

Reply:
Will do.

I'm not sure if `INSTALL_MOD_PATH=$DESTDIR` will work for other distributions, but `INSTALL_MOD_PATH=$DESTDIR/usr MODLIB=...` ends up with a broken image (the image builds but it won't boot -- the start job for `/dev/disk/...` times out) because everything is installed in `/usr/usr/...`.